include/annotated.hpp

```diff
@@ -0,0 +1,189 @@
+#ifndef GRAPH_PROTOTYPE_ANNOTATED_HPP
+#define GRAPH_PROTOTYPE_ANNOTATED_HPP
+
+#include <string_view>
+#include <type_traits>
+#include <utils.hpp>
+
+namespace fair::graph {
+
+/**
+ * @brief a template wrapping structure, which holds a static documentation (e.g. mark down) string as its value.
+ * It's used as a trait class to annotate other template classes (e.g. blocks or fields).
+ */
+template<fair::meta::fixed_string doc_string>
+struct Doc {
+    static constexpr fair::meta::fixed_string value = doc_string;
+};
+
+using EmptyDoc = Doc<"">; // nomen-est-omen
+
+template<typename T>
+struct is_doc : std::false_type {};
+
+template<fair::meta::fixed_string N>
+struct is_doc<Doc<N>> : std::true_type {};
+
+template<typename T>
+concept Documentation = is_doc<T>::value;
+
+/**
+ * @brief Unit is a template structure, which holds a static physical-unit (i.e. SI unit) string as its value.
+ * It's used as a trait class to annotate other template classes (e.g. blocks or fields).
+ */
+template<fair::meta::fixed_string doc_string>
+struct Unit {
+    static constexpr fair::meta::fixed_string value = doc_string;
+};
+
+using EmptyUnit = Unit<"">; // nomen-est-omen
+
+template<typename T>
+struct is_unit : std::false_type {};
+
+template<fair::meta::fixed_string N>
+struct is_unit<Unit<N>> : std::true_type {};
+
+template<typename T>
+concept UnitType = is_unit<T>::value;
+
+static_assert(Documentation<EmptyDoc>);
+static_assert(UnitType<EmptyUnit>);
+static_assert(!UnitType<EmptyDoc>);
+static_assert(!Documentation<EmptyUnit>);
+
+/**
+ * @brief Annotates field etc. that the entity is visible.
+ */
+struct Visible {};
+
+/**
+ * @brief Annotates node, indicating to calling schedulers that it may block due IO.
+ */
+struct BlockingIO {};
+
+/**
+ * @brief Annotates templated node, indicating which port data types are supported.
+ */
+template<typename... Ts>
+struct SupportedTypes {};
+
```

Comment:
The approach of adding supported type to node's parameter list has a serious limitation that I realized just yesterday working on the plugin system - it is not accessible until a node is instantiated. But we can't create a node as we don't know the `T` for which we can create it.

Reply:
need to think a bit more about this ... :thinking: